### PR TITLE
Issue #26: Getting an Azure resource by full ID

### LINKED ISSUE #26: Node initialisation hangs if Azure VM ID is used in the networks config
If a `cluster.networks.config` preference contains full ID of the image, e.g. `/subscriptions/11111111-2222-2222-3333-4444444/resourceGroups/res-grp/providers/Microsoft.Compute/images/CloudPipeline-Image-Common` (which is generated by the `build_infra.sh` script) - `InitializeNode` task hangs (almost) forever. For it was 20+ minutes.

Last log:
image

This leads me the idea that `nodeup` for azure does not contain enough error handling.
So we shall:
1. Add more logging - if something may fail: catch it and report
2. If something failed - drop the `nodeup` script execution
3. If something may hang - introduce wait loops

### PR DESCRIPTION
**Pull request by SilinPavel**
### Summary
This PR resolves #26 and #25 .
Now we can call azure resources by full Azure ID form like: `/subscriptions/<sub>/resourceGroups/<res_grp>/providers/Microsoft.<service>/<res_type>/<resource>`
I also added here commit that resolves issue #25, since this two issues are very close and touch the same script

### CODE REVIEW

**Comment by sidoruka** [comment]
@SilinPavel , am I correct that:
1. #26 will now accept both short and full notations?
2. #25 - not clear on the implementation, how is it handled now?

**Comment by SilinPavel** [comment]
@sidoruka 
1. Yes, it will accept <res_grp>/<res_name> and /subscriptions/<sub-id>/resourceGroups/<res-grp>/providers/<provider>/<type>/<resource> forms
2. Now it is handled this way: if script cannot find any of network from a network list, script tries to find any of available network from the same region and the same resource group in the subscription. If there is nothing to find, script falls with message `No networks with subnet found for location: {region} in resourceGroup: {res_group}`

File: scripts/autoscaling/azure/nodeup.py [hunk 1]
```diff
@@ -673,6 +670,33 @@ def get_user_data_script(cloud_region, ins_type, ins_img, kube_ip, kubeadm_token
         raise RuntimeError('Unable to get init.sh path')
 
 
+def get_res_grp_and_res_name_from_string(resource_id, resource_type):
+    resource_params = resource_id.split("/")
+    if len(resource_params) == 2:
+        resource_group, resource = resource_params[0], resource_params[1]
+    # according to full ID form: /subscriptions/<sub-id>/resourceGroups/<res-grp>/providers/Microsoft.Compute/images/<image>
+    elif len(resource_params) == 9 and resource_params[3] == 'resourceGroups' and resource_params[7] == resource_type:
```
[the remark this answers is not in the document]
Comment (SilinPavel) [reply scripts/autoscaling/azure/nodeup.py]: done

File: scripts/autoscaling/azure/nodeup.py [hunk 2]
```diff
@@ -673,6 +670,33 @@ def get_user_data_script(cloud_region, ins_type, ins_img, kube_ip, kubeadm_token
...
+            "Node Up process will be stopped.")
+    return resource_group, resource
+
+
+def get_subnet_name_from_id(subnet_id):
+    if "/" not in subnet_id:
+        return subnet_id
+    subnet_params = subnet_id.split("/")
+    # according to /subscriptions/<sub>/resourceGroups/<res_grp>/providers/Microsoft.Network/virtualNetworks/<vnet>/subnets/<subnet>
+    if len(subnet_params) == 11 and subnet_params[3] == "resourceGroups" and subnet_params[9] == "subnets":
```
[the remark this answers is not in the document]
Comment (SilinPavel) [reply scripts/autoscaling/azure/nodeup.py]: @mzueva done

**Pull request merged**